app/controllers/case_logs_controller.rb

```diff
@@ -122,13 +122,22 @@ def find_resource
 
   def filtered_case_logs
     user_case_logs = current_user.case_logs
-    status_filter = JSON.parse(session[:case_logs_filters])["status"] if session[:case_logs_filters].present?
-    return user_case_logs unless status_filter
+    if session[:case_logs_filters].present?
+      filters = JSON.parse(session[:case_logs_filters])
 
-    user_case_logs.filter_by_status(status_filter)
+      %w[status year].each do |category|
+        if filters[category]
+          filtered_by_category = filters[category].select(&:present?).map { |filter| user_case_logs.public_send("filter_by_#{category}", filter) }.flatten
+          user_case_logs = CaseLog.where(id: filtered_by_category.map(&:id))
```

Comment:
Does this work for combined filters? Aren't you just setting the returned collection to whichever filter category was last run?

Also I think you want to build up the scopes (they're deferred and composable) so that you ultimately just run a single DB query to get the result rather than running them sequentially.